### PR TITLE
fix(Transfer): clicking the checkbox position cannot be unchecked and onSelectChange is triggered twice

### PR DESCRIPTION
**Pull request by edc-hui**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [X] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
#39030

### 💡 Background and solution
该bug由PR #37973 造成, 故移除#37973的代码，并采用其他方式修复#37973想要解决的checkbox 在hover时样式问题

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     fix Transfer clicking the checkbox position cannot be unchecked and onSelectChange is triggered twice      |
| 🇨🇳 Chinese |      修复 Transfer 组件 点击复选框位置不可以取消选中，并触发了两次onSelectChange 问题    |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

### CODE REVIEW

File: components/checkbox/style/index.tsx [hunk 1]
```diff
@@ -208,6 +206,7 @@ export const genCheckboxStyle: GenerateStyle<CheckboxToken> = (token) => {
           animationTimingFunction: 'ease-in-out',
           animationFillMode: 'backwards',
           content: '""',
+          transition: `all ${token.motionDurationSlow}`,
```
Comment (MadCcc) [inline components/checkbox/style/index.tsx:209]:
```suggestion
```

**Comment by MadCcc** [comment]
第一个 demo 还是有问题，看起来是非受控时 wrapper 没有 checked 
image
image

**Comment by edc-hui** [comment]
> 第一个 demo 还是有问题，看起来是非受控时 wrapper 没有 checked。antd 里没有非受控状态，需要额外加点 css 了 image image

我再测试下看看

File: components/checkbox/style/index.tsx (hunk 1 above)
Comment (edc-hui) [reply components/checkbox/style/index.tsx:219]: 您的这个修改建议：我提下我的意见，你看看对不对。
受控状态下:   .ant-checkbox-checked:after 的 transition 时间应该和 ant-checkbox-inner 的transition时间保持一致才对，使用motionDurationSlow，因hover的时候是 ant-checkbox-inner 变背景色，ant-checkbox-inner的transition时间用的是motionDurationSlow，   故时间应该和 ant-checkbox-inner 的transition时间保持一致。

image

**Comment by edc-hui** [comment]
@MadCcc 我添加了css，目前我自测非受控状态下Checkbox在hove的时候 表现没问题了

**Review by MadCcc — APPROVED** [review]

**Comment by pepe95270** [comment]
Hello,
The checkboxes are still buggy, try clicking on the top limit of the checkbox and the bug is still happening :/

**Comment by edc-hui** [comment]
> Hello, The checkboxes are still buggy, try clicking on the top limit of the checkbox and the bug is still happening :/

i will fix the problem

**Pull request merged**